### PR TITLE
chgres_cube: Implement WMO grib2 template 1 (rotated lat-lon) read capability

### LINKED ISSUE #901: Add chgres_cube capability to read data on WMO rotated lat-lon grid (template 1)
The RRFS fire weather nest will be implemented operationally alongside RRFSv1, and is designed to replicate the current capabilities of the NAM fire weather nest.  RRFSFW will be initialized off of RRFS grib2 data, specifically the rotated lat-lon grib2 files.  Those files are on grid template 1 (WMO rotated lat-lon) and chgres_cube needs to be updated to process those files correctly.

Full completion of this capability will also depend on addressing #850 . For now, we can test with a sample subset provided by 
@BenjaminBlake-NOAA who requested this feature.

### PR DESCRIPTION
**Pull request by LarissaReames**
## DESCRIPTION OF CHANGES: 
RRFS Fire needs to be able to initialize from the rotated lat-lon grib2 RRFS files which are on the WMO grib2 template 1. This implements the capacity to read data on that template. Also added 'RRFS' as a supported _external_model_ namelist input. though it's not required to support this grid template. Full versions of those RRFS file, however, are still not supported as they are too large (>2GB). That will be addressed when #850 is fixed. 

## TESTS CONDUCTED: 
Intel builds on Jet and Hera. Gnu builds on Hera. Unit tests pass on Hera. Consistency tests pass on Hera. Need support for: Orion, Hercules, WCOSS2 intel builds. Debug build on WCOSS2.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera, Hercules and WCOSS2). Done using 804ea27.
- [X] Compile branch on Hera using GNU. Done using 804ea27.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done on Dogwood using 804ea27.
- [X] Run unit tests locally on any Tier 1 machine. Done on Hera using 804ea27.
- [X] Run relevant consistency tests locally on all Tier 1 machine. Done using 804ea27. All tests failed on Jet. However, the baseline was created on xjet. Those nodes have been migrated to a new operating system, on which UFS_UTILS has not been tested. Running on vjet nodes resulted in floating point differences.

## DEPENDENCIES:

N/A

## DOCUMENTATION:
Updated documentation to note that 'RRFS' is now a supported _external_model_ option. 

## ISSUE: 
Resolves #901 

## CONTRIBUTORS (optional): 
@GeorgeGayno-NOAA contributed the grib1 grid definition information

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
Should we add a new consistency test? This test uses the local NCEP template for rotated latl/on:
- https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/chgres_cube/13km.conus.rap.grib2.sh

The new test could be the same except for the input data (which would use the WMO template).

**Comment by LarissaReames** [comment]
> Should we add a new consistency test? This test uses the local NCEP template for rotated latl/on:
> 
> * https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/chgres_cube/13km.conus.rap.grib2.sh
> 
> The new test could be the same except for the input data (which would use the WMO template).

I'd thought about that, but thought maybe we'd wait until we can support the full RRFS grib2 files with the new g2 library. I was only able to confirm that the current updates worked with a sub-CONUS file from Ben interpolated to an even smaller sub-CONUS output grid.

File: sorc/chgres_cube.fd/atm_input_data.F90 [hunk 1]
```diff
@@ -3018,6 +3018,14 @@ subroutine read_winds(u,v,localpet,octet_23,rlevs,lugb,pdt_num)
      print*, "- CALL CALCALPHA_ROTLATLON with center lat,lon = ",latin1,lov
      call calcalpha_rotlatlon(lat,lon,latin1,lov,alpha)
 
+   elseif (gfld%igdtnum == 1) then ! grid definition template number - non-E stagger rotated lat/lon grid
+
+     latin1 = real(float(gfld%igdtmpl(20))/1.0E6, kind=esmf_kind_r4)
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/atm_input_data.F90]: I would double check how `latin1` is set. In template 32769 it is the latitude of the center of the grid. In template 1, it is the latitude of the south pole of projection. In `model_grid`, I had to add 90 degrees.

File: sorc/chgres_cube.fd/model_grid.F90 [hunk 2]
```diff
@@ -1528,6 +1532,112 @@ subroutine gdt_to_gds(igdtnum, igdstmpl, igdtlen, kgds, ni, nj, res)
...
+     kgds(6)=0                      ! oct 17, resolution and component flags
+     if (igdstmpl(1)==2 ) kgds(6)=64
+     if ( btest(igdstmpl(14),4).OR.btest(igdstmpl(14),5) )  kgds(6)=kgds(6)+128
+     if ( btest(igdstmpl(14),3) ) kgds(6)=kgds(6)+8
+
+     kgds(7)=nint(float(igdstmpl(20))/float(iscale)*1000.)       ! octs 18-20,
+                                                                 ! Lat of cent of rotation
+     kgds(8)=nint(float(igdstmpl(21))/float(iscale)*1000.)       ! octs 21-23,
+     print*, "INPUT LAT, LON CENTER ", kgds(7), kgds(8) 
+                                                                 ! Lon of cent of rotation
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/model_grid.F90:1560]: Note here how I had to add 90 degrees to get the longitude of the grid center.

Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/model_grid.F90:1560]: Sorry, I mean latitude.

File: sorc/chgres_cube.fd/atm_input_data.F90 (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/atm_input_data.F90]: Sorry, I meant `latin1`

File: sorc/chgres_cube.fd/model_grid.F90 (hunk 2 above)
Comment (LarissaReames) [reply sorc/chgres_cube.fd/model_grid.F90:1560]: Gah, yes I was checking against that print out but I missed that next line where you add 90 degrees. I will make that modification.

File: sorc/chgres_cube.fd/atm_input_data.F90 (hunk 1 above)
Comment (LarissaReames) [reply sorc/chgres_cube.fd/atm_input_data.F90]: Fixed.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
> > Should we add a new consistency test? This test uses the local NCEP template for rotated latl/on:
> > 
> > * https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/chgres_cube/13km.conus.rap.grib2.sh
> > 
> > The new test could be the same except for the input data (which would use the WMO template).
> 
> I'd thought about that, but thought maybe we'd wait until we can support the full RRFS grib2 files with the new g2 library. I was only able to confirm that the current updates worked with a sub-CONUS file from Ben interpolated to an even smaller sub-CONUS output grid.

Did you test the coldstart files in the model? 

I will help with testing on the machines you don't have access to.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA Can I merge this?

**Comment by LarissaReames** [comment]
@GeorgeGayno-NOAA sounds good to me!

**Pull request merged**